apollo-ios-codegen/Sources/IR/IR+NamedFragment.swift

```diff
@@ -22,6 +22,23 @@ public class NamedFragment: Definition, Hashable, CustomDebugStringConvertible {
   public var name: String { definition.name }
   public var type: GraphQLCompositeType { definition.type }
   public var isLocalCacheMutation: Bool { definition.isLocalCacheMutation }
+  
+  /// Indicates if the parent type has a single keyField named `id`.
+  public var isIdentifiable: Bool {
+    guard definition.selectionSet.selections.contains(where: {
+      guard case .field(let field) = $0 else { return false }
+      return field.name == "id"
+    }) else { return false }
+    
+    switch(type) {
```

Comment:
I think we can reduce the duplication here with the selection set check. I tried to push a PR to merge into this one but fork permissions don't allow that so I'm going to add it as a commit to this PR.

Reply:
https://github.com/apollographql/apollo-ios-dev/pull/595/commits/c21c6838105acae842ca1fc28082dc5475d9c837